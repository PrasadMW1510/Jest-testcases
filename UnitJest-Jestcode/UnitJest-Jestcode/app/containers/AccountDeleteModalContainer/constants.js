/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
