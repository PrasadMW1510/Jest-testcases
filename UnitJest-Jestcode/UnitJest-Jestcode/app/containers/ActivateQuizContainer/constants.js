/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
