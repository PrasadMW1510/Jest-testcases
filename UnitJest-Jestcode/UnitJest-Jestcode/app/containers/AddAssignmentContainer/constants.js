/*
 *
 * AddAssignmentContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/AddAssignmentContainer/DEFAULT_ACTION';
export const GET_STUDENTDETAIL_SUCCESS = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENTDETAIL_FAILURE = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_STUDENT_DETAILS = 'app/AddAssignmentContainer/GET_STUDENT_DETAILS';
export const POST_NEW_ASSIGNMENT = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT';
export const POST_NEW_ASSIGNMENT_SUCCESS = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS';
