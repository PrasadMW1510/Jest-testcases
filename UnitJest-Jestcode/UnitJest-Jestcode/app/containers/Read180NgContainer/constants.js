/*
 *
 * Read180NgContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/Read180NgContainer/DEFAULT_ACTION';
export const READ_180_NG_LIST = 'app/Read180ngContainer/DEFAULT_ACTION';
export const READ_180_NG_DATA_REQUEST = 'containers/Read180NgContainer/Read180NG_detail_request';
export const READ_180_NG_DATA_REQUEST_SUCCESS =
  'containers/Read180NgContainer/Read180NG_detail_request_success';
export const SET_READ_180_NG_DATA = 'containers/Read180NgContainer/set_read180NG_data';
export const SET_180_NG_DATA_REQUEST_SUCCESS =
  'container/Read180NgContainer/set_read180NG_data_success';
export const DELETE_ASSIGNMENT_DATA = 'container/Read180NgContainer/delete_assignment_data';
export const UNINITIALIZED_ITEM_COUNT = 'container/Read180NgContainer/uninitialized_item_count';
