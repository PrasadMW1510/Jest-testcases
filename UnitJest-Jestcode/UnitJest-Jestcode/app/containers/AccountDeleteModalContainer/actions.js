/*
 *
 * AccountDeleteModalContainer actions
 *
 */
import { generalFailure } from 'containers/App/actions';
import * as Constants from './constants';

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteRequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete action
 *
 * @param accountDeletePayload
 * @returns {{type}}
 */
export function postAccountDeleteMIARequest(accountDeletePayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_DELETE_MIA_REQUEST,
    payload: accountDeletePayload,
  };
}

/**
 * Handles API logic for account delete is a success
 *
 *  @returns {{type}}
 */
export function postAccountDeleteRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account delete is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountDeleteRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_DELETE_REQUEST_FAILURE,
    error,
  });
}

/**
 * Handles API logic for account unenroll action
 *
 * @param accountUnenrollPayload
 * @returns {{type}}
 */
export function postAccountUnenrollRequest(accountUnenrollPayload) {
  // payload contains (1)list of accts to delete, (2) searchOpts used to refresh
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST,
    payload: accountUnenrollPayload,
  };
}

/**
 * Handles API logic for account unenroll is a success
 *
 *  @returns {{type}}
 */
export function postAccountUnenrollRequestSuccess() {
  return {
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS,
  };
}

/**
 * Handles API logic for when account unenroll is a failure
 *
 * @param error
 * @returns {{type, error: *}}
 */
export function postAccountUnenrollRequestFailure(error) {
  return generalFailure({
    type: Constants.POST_ACCOUNT_UNENROLL_REQUEST_FAILURE,
    error,
  });
}
