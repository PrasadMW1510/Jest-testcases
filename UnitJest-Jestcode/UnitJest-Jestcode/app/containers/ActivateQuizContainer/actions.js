/*
 *
 * ActivateQuizContainer actions
 *
 */

import * as Constants from './constants';

export function activateQuizRequest(activateQuizPreviewData, searchOpts, searchDataChecked) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST,
    activateQuizPreviewData,
    searchOpts,
    searchDataChecked,
  };
}

export function getPostActivateRequestFailure(error) {
  return {
    type: Constants.ACTIVATE_QUIZ_LIST_FAILURE,
    error,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}

export function initialiseModalFlag() {
  return {
    type: Constants.INITIALIZE_MODAL_FLAG,
  };
}
