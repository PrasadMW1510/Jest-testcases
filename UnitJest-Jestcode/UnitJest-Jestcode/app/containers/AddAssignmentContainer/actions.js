/*
 *
 * AddAssignmentContainer actions
 *
 */

import * as Constants from './constants';

export function getStudentDetails(data) {
  return {
    type: Constants.GET_STUDENT_DETAILS,
    data,
  };
}
export function getStudentDetailsSuccess(data) {
  return {
    type: Constants.GET_STUDENTDETAIL_SUCCESS,
    data,
  };
}

export function getStudentDetailsFailure(error) {
  return {
    type: Constants.GET_STUDENTDETAIL_FAILURE,
    error,
  };
}

export function postSaveNewAssignment(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT,
    data,
  };
}

export function postSaveNewAssignmentSuccess(data) {
  return {
    type: Constants.POST_NEW_ASSIGNMENT_SUCCESS,
    data,
  };
}

export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
