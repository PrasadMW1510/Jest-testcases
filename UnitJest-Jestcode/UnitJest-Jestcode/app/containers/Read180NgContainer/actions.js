/*
 *
 * Read180NgContainer actions
 *
 */
import * as Constants from './constants';

export function read180ngRequest(read180ngPreviewData) {
  return {
    type: Constants.READ_180_NG_LIST,
    read180ngPreviewData,
  };
}

export function getRead180DataRequest(read180nData) {
  return {
    type: Constants.READ_180_NG_DATA_REQUEST,
    read180nData,
  };
}

export function getRead180DataRequestSuccess(resultsData) {
  return {
    type: Constants.READ_180_NG_DATA_REQUEST_SUCCESS,
    resultsData,
  };
}

export function setRead180NgData(read180ngData, workItemId) {
  return {
    type: Constants.SET_READ_180_NG_DATA,
    read180ngData,
    workItemId,
  };
}

export function setRead180DataRequestSuccess(resultsData) {
  return {
    type: Constants.SET_180_NG_DATA_REQUEST_SUCCESS,
    resultsData,
  };
}
export function deleteAssignmentData(Data) {
  return {
    type: Constants.DELETE_ASSIGNMENT_DATA,
    Data,
  };
}
export function defaultAction() {
  return {
    type: Constants.DEFAULT_ACTION,
  };
}
