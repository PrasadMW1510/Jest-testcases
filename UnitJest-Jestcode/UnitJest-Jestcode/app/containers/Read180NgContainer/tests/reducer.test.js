import { fromJS } from 'immutable';
import read180NgContainerReducer from '../reducer';
import * as Actions from '../actions';
import * as Constants from '../constants';

describe('Read180NgContainer reducer', () => {
  const initialState = fromJS({
    read180Program: {
      loading: false,
      results: [],
      itemCount: Constants.UNINITIALIZED_ITEM_COUNT,
      paginationData: {},
    },
    read180PostProgram: {
      loading: false,
      results: [],
      itemCount: Constants.UNINITIALIZED_ITEM_COUNT,
      paginationData: {},
    },
  });

  it('returns the initial state', () => {
    expect(read180NgContainerReducer(undefined, {})).toEqual(fromJS(initialState));
  });

  it('should handle READ_180_NG_DATA_REQUEST_SUCCESS', () => {
    const searchResultsobj = {
      output_data: [],
      pagination_data: [{}],
      item_count: [-1],
    };
    const updatedState = fromJS({
      read180Program: {
        loading: false,
        results: undefined,
        itemCount: -1,
        paginationData: {},
      },
      read180PostProgram: {
        loading: false,
        results: [],
        itemCount: Constants.UNINITIALIZED_ITEM_COUNT,
        paginationData: {},
      },
    });
    expect(
      read180NgContainerReducer(undefined, Actions.getRead180DataRequestSuccess(searchResultsobj))
    ).toEqual(updatedState);
  });

  it('should handle SET_180_NG_DATA_REQUEST_SUCCESS', () => {
    const searchResultsobj = {
      output_data: [],
      pagination_data: [{}],
      item_count: [-1],
    };
    const updatedState = fromJS({
      read180Program: {
        loading: false,
        results: [],
        itemCount: Constants.UNINITIALIZED_ITEM_COUNT,
        paginationData: {},
      },
      read180PostProgram: {
        loading: false,
        results: undefined,
        itemCount: -1,
        paginationData: {},
      },
    });
    expect(
      read180NgContainerReducer(undefined, Actions.setRead180DataRequestSuccess(searchResultsobj))
    ).toEqual(updatedState);
  });

  it('should handle DEFAULT_ACTION', () => {
    expect(read180NgContainerReducer(undefined, Actions.defaultAction())).toMatchSnapshot();
  });
});
