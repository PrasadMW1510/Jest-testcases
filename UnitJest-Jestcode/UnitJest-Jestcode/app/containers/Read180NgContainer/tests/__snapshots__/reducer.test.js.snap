// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read180NgContainer reducer should handle DEFAULT_ACTION 1`] = `
Immutable.Map {
  read180Program: Immutable.Map {
    loading: false,
    results: Immutable.List [
    ],
    itemCount: "container/Read180NgContainer/uninitialized_item_count",
    paginationData: Immutable.Map {
    },
  },
  read180PostProgram: Immutable.Map {
    loading: false,
    results: Immutable.List [
    ],
    itemCount: "container/Read180NgContainer/uninitialized_item_count",
    paginationData: Immutable.Map {
    },
  },
}
`;
